src/test/scala/vsys/blockchain/state/opcdiffs/BasicOpcDiffTest.scala

```diff
@@ -0,0 +1,126 @@
+      DataEntry(Longs.toByteArray(1), DataType.Amount),
+      DataEntry(Longs.toByteArray(1), DataType.Amount),
+      Seq.empty, 0) should be (Right(Seq(DataEntry(Longs.toByteArray(1), DataType.Amount))))
+    BasicOpcDiff.divide(
+      DataEntry(Longs.toByteArray(1), DataType.Amount),
+      DataEntry(Longs.toByteArray(1), DataType.Timestamp),
+      Seq.empty, 0) should be (Left(ContractDataTypeMismatch))
+    BasicOpcDiff.divide(
+      DataEntry(Longs.toByteArray(1), DataType.Amount),
+      DataEntry(Longs.toByteArray(0), DataType.Amount),
+      Seq.empty, 0) should be (Left(ValidationError.OverflowError))
+    BasicOpcDiff.divide(
+      DataEntry(Longs.toByteArray(1), DataType.ShortText),
+      DataEntry(Longs.toByteArray(1), DataType.ShortText),
+      Seq.empty, 0) should be (Left(ContractUnsupportedOPC))
+
+    BasicOpcDiff.minimum(
+      DataEntry(Longs.toByteArray(1), DataType.Amount),
+      DataEntry(Longs.toByteArray(0), DataType.Amount),
+      Seq.empty, 0) should be (Right(Seq(DataEntry(Longs.toByteArray(0), DataType.Amount))))
+    BasicOpcDiff.minimum(
+      DataEntry(Longs.toByteArray(1), DataType.Amount),
+      DataEntry(Longs.toByteArray(1), DataType.Timestamp),
+      Seq.empty, 0) should be (Left(ContractDataTypeMismatch))
+    BasicOpcDiff.minimum(
+      DataEntry(Longs.toByteArray(1), DataType.ShortText),
+      DataEntry(Longs.toByteArray(1), DataType.ShortText),
+      Seq.empty, 0) should be (Left(ContractUnsupportedOPC))
+
+    BasicOpcDiff.maximum(
+      DataEntry(Longs.toByteArray(1), DataType.Amount),
+      DataEntry(Longs.toByteArray(0), DataType.Amount),
+      Seq.empty, 0) should be (Right(Seq(DataEntry(Longs.toByteArray(1), DataType.Amount))))
+    BasicOpcDiff.maximum(
+      DataEntry(Longs.toByteArray(1), DataType.Amount),
+      DataEntry(Longs.toByteArray(1), DataType.Timestamp),
+      Seq.empty, 0) should be (Left(ContractDataTypeMismatch))
+    BasicOpcDiff.maximum(
+      DataEntry(Longs.toByteArray(1), DataType.ShortText),
+      DataEntry(Longs.toByteArray(1), DataType.ShortText),
+      Seq.empty, 0) should be (Left(ContractUnsupportedOPC))
+
+    BasicOpcDiff.concat(
+      DataEntry(Longs.toByteArray(1), DataType.Amount),
+      DataEntry(Longs.toByteArray(1), DataType.Amount),
+      Seq.empty, 0) should be (
+      Right(Seq(DataEntry.create(Longs.toByteArray(1) ++ Longs.toByteArray(1),
+        DataType.ShortBytes).right.get)))
+    BasicOpcDiff.concat(
+      DataEntry(Array.fill(maxShortBytesLength){0}, DataType.ShortBytes),
+      DataEntry(Array.fill(1){0}, DataType.ShortBytes),
+      Seq.empty, 0) should be (Left(InvalidDataEntry))
+
+    BasicOpcDiff.constantGet(Array(DataType.Amount.id.toByte) ++ Longs.toByteArray(1),
+      Seq.empty, 0) should be (Right(Seq(DataEntry(Longs.toByteArray(1), DataType.Amount))))
+    BasicOpcDiff.constantGet(Array(DataType.Amount.id.toByte) ++ Ints.toByteArray(1),
+      Seq.empty, 0) should be (Left(InvalidDataEntry))
+  }
```

Comment:
We need to add `pointer` in the original code and test the usage here.

Reply:
`pointer` in the original code? What is that?

Reply:
in `BasicOpcDiff.scala`, we may check the pointer range, then test the incorrect pointer here.

Reply:
yea, I checked when the pointer is out of range